otcextensions/osclient/nat/v2/dnat.py

```diff
@@ -0,0 +1,252 @@
+
+
+class CreateDnatRule(command.ShowOne):
+    _description = _("Create new DNAT Rule")
+
+    def get_parser(self, prog_name):
+        parser = super(CreateDnatRule, self).get_parser(prog_name)
+        parser.add_argument(
+            '--nat-gateway-id',
+            required=True,
+            metavar="<nat_gateway_id>",
+            help=_("Specifies the ID of the NAT gateway"))
+        parser.add_argument(
+            '--port-id',
+            metavar='<port_id>',
+            help=_('Specifies the port ID of an ECS or a BMS.'))
+        parser.add_argument(
+            '--private-ip',
+            metavar='<private_ip>',
+            help=_('Specifies the private IP address, for example, '
+                   'the IP address of a Direct Connect connection.'))
+        parser.add_argument(
+            '--internal-service-port',
+            metavar='<internal_service_port>',
+            required=True,
+            help=_('Specifies port used by ECSs or BMSs toprovide '
+                   'services for external systems.'))
+        parser.add_argument(
+            '--floating-ip-id',
+            metavar="<floating_ip_id>",
+            required=True,
+            help=_('Specifies the EIP ID. Multiple EIPs are '
+                   'separated using commas'))
+        parser.add_argument(
+            '--external-service-port',
+            metavar='<external_service_port>',
+            required=True,
+            help=_('Specifies the port for providing external services.'))
+        parser.add_argument(
+            '--protocol',
+            metavar='<protocol>',
+            required=True,
+            help=_('Specifies the protocol type.'))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.nat
+
+        args_list = [
+            'nat_gateway_id',
+            'port_id',
+            'private_ip',
+            'internal_service_port',
+            'floating_ip_id',
+            'external_service_port',
+            'protocol'
+        ]
+        attrs = {}
+        for arg in args_list:
+            if getattr(parsed_args, arg):
+                attrs[arg] = getattr(parsed_args, arg)
+
+        obj = client.create_dnat_rule(**attrs)
+
+        display_columns, columns = _get_columns(obj)
+        data = utils.get_item_properties(obj, columns)
+
+        return (display_columns, data)
+
+
+class DeleteDnatRule(command.Command):
+
+    _description = _("Deletes Dnat Rule(s).")
+
+    def get_parser(self, prog_name):
+        parser = super(DeleteDnatRule, self).get_parser(prog_name)
+        parser.add_argument(
+            'dnat',
+            metavar='<dnat_id>',
+            help=_('Specifies the ID of the DNAT Rule'),
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.nat
+        dnat_rule = client.get_dnat_rule(parsed_args.dnat)
+        client.delete_dnat_rule(dnat_rule.id)
```

Comment:
delete non existent rule?

Reply:
It will return error
```
openstack nat dnat rule delete abc-abc-abc-abc
No Dnat found for abc-abc-abc-abc: Client Error for url: https://nat.eu-de.otc.t-systems.com/v2.0/dnat_rules/abc-abc-abc-abc, Not Found
```